loadtest/control/simulcontroller/controller.go

```diff
@@ -48,103 +50,108 @@ func (c *SimulController) Run() {
 
 	c.status <- control.UserStatus{ControllerId: c.id, User: c.user, Info: "user started", Code: control.USER_STATUS_STARTED}
 
-	defer c.sendStopStatus()
-
-	if resp := control.SignUp(c.user); resp.Err != nil {
-		c.status <- c.newErrorStatus(resp.Err)
-	} else {
-		c.status <- c.newInfoStatus(resp.Info)
+	initActions := []userAction{
+		{
+			run: control.SignUp,
+		},
+		{
+			run: func(u user.User) control.UserActionResponse {
+				return c.login()
+			},
+		},
+		{
+			run: func(u user.User) control.UserActionResponse {
+				return c.joinTeam(u)
+			},
+		},
 	}
 
-	if resp := control.Login(c.user); resp.Err != nil {
-		c.status <- c.newErrorStatus(resp.Err)
-	} else {
-		c.status <- c.newInfoStatus(resp.Info)
-		c.connect()
-	}
+	defer func() {
+		if err := c.user.Disconnect(); err != nil {
+			c.status <- c.newErrorStatus(err)
+		}
+		c.user.Cleanup()
+		c.sendStopStatus()
+		close(c.stopped)
+	}()
 
-	if resp := c.reload(false); resp.Err != nil {
-		c.status <- c.newErrorStatus(resp.Err)
-	} else {
-		c.status <- c.newInfoStatus(resp.Info)
+	for _, action := range initActions {
+		if resp := action.run(c.user); resp.Err != nil {
+			c.status <- c.newErrorStatus(resp.Err)
+		} else {
+			c.status <- c.newInfoStatus(resp.Info)
+		}
+		select {
+		case <-c.stop:
+			return
+		default:
+		}
 	}
 
-	if resp := c.joinTeam(c.user); resp.Err != nil {
-		c.status <- c.newErrorStatus(resp.Err)
-	} else {
-		c.status <- c.newInfoStatus(resp.Info)
-	}
+	go func() {
+		for {
+			select {
+			case <-time.After(60 * time.Second):
```

Comment:
Ah the infamous `getUsersStatusesByIds`. Is 60 secs the duration in webapp too ? Let's refactor this to a constant naming it `getUsersStatusByIdsInterval`.

I would suggest to move this to a different method in a new file altogether, that can house all the activities that the webapp performs periodically. Name it something like periodicActions. And here, we can just do `go c.periodicActions()`. We can keep adding new stuff in the for-select loop.

Reply:
It looks like it:

https://github.com/mattermost/mattermost-redux/blob/6fb382b516d2769e1c05752b80f2a9b860a5c90d/src/constants/general.ts#L15

Agree on making a constant and on having a separate goroutine to handle periodic actions.